PySDM/backends/impl_numba/methods/freezing_methods.py

```diff
@@ -92,6 +95,48 @@ def freeze_time_dependent_body(  # pylint: disable=too-many-arguments
 
         self.freeze_time_dependent_body = freeze_time_dependent_body
 
+        j_hom = self.formulae.homogeneous_ice_nucleation_rate.j_hom
+
+        @numba.njit(**self.default_jit_flags)
+        def freeze_time_dependent_homogeneous_body(  # pylint: disable=unused-argument,too-many-arguments
+            rand,
+            attributes,
+            timestep,
+            cell,
+            a_w_ice,
+            temperature,
+            relative_humidity_ice,
+            thaw,
+        ):
+
+            n_sd = len(attributes.signed_water_mass)
+            for i in numba.prange(n_sd):  # pylint: disable=not-an-iterable
+                cell_id = cell[i]
+                if thaw and frozen_and_above_freezing_point(
+                    attributes.signed_water_mass[i], temperature[cell_id]
+                ):
+                    _thaw(attributes.signed_water_mass, i)
+                elif unfrozen_and_ice_saturated(
+                    attributes.signed_water_mass[i], relative_humidity_ice[cell_id]
+                ):
+                    d_a_w_ice = (relative_humidity_ice[cell_id] - 1.0) * a_w_ice[
+                        cell_id
+                    ]
+                    if 0.23 < d_a_w_ice < 0.34:
```

Comment:
@tluettm, these constants would best be moved to `constants_default` and referred from a new routine in `physics.homogeneous_ice_nucleation_rate` - we could add a new function there like `d_a_w_ice_within_range`. However, I'm puzzled why beyond this range freezing is effectively disabled? In Spichtinger et al. 2023, in the passage under Fig.  11, it is suggested to use a constant J_hom above the maximal d_a_w for a given parameterisation.